packages/dds/tree/src/shared-tree/sharedTree.ts

```diff
@@ -95,16 +94,49 @@ export interface ISharedTreeCheckout extends AnchorLocator {
 	/**
 	 * Current contents.
 	 * Updated by edits (local and remote).
-	 * Use `runTransaction` to create a local edit.
+	 * Use `runTransaction` to create a local edit. TODO
 	 */
 	readonly forest: IForestSubscription;
 
+	/**
+	 * Used to edit the state of the tree. Edits will be immediately applied locally to the tree.
+	 * If there is no transaction currently ongoing, then the edits will be submitted to Fluid immediately as well.
+	 */
+	readonly editor: IDefaultEditBuilder;
+
+	/**
+	 * An collection of functions for managing transactions. Transactions allow edits to be batched into atomic units.
+	 * Edits made during a transaction will update the local state of the tree immediately, but will be squashed into a single edit
+	 * when the transaction is committed. If the transaction is aborted, the local state will be reset to what it was before
+	 * the transaction began. Transactions may nest, meaning that a transaction may be started while a transaction is already
+	 * ongoing.
+	 *
+	 * TODO: Provide an option to avoid updating local state during a transaction.
+	 */
+	readonly transaction: {
+		/**
+		 * Start a new transaction. If a transaction is already in progress when this new transaction starts, then it will remain in
+		 * progress when the new transaction is committed or aborted.
```

Comment:
This could maybe be a bit clearer in describing that the new transaction will be nested in the current transaction.